protoc-gen-swagger/genswagger/template.go

```diff
@@ -1448,6 +1459,88 @@ func processExtensions(inputExts map[string]*structpb.Value) ([]extension, error
 	return exts, nil
 }
 
+func validateHeaderType(headerType string) error {
+	switch headerType {
+	case
+		"string",
+		"number",
+		"integer",
+		"boolean":
+		return nil
+	}
+	return fmt.Errorf("the provided header type %q is not supported", headerType)
+}
+
+func validateDefaultValueType(headerType string, defaultValue string) error {
+	switch headerType {
+	case
+		"string":
+		if !(isString(defaultValue)) {
+			return fmt.Errorf("the provided default value %q does not match provider type %q, or is not properly quoted with escaped quotations", defaultValue, headerType)
+		}
+	case
+		"number":
+		_, err := strconv.ParseFloat(defaultValue, 64)
+		if err != nil {
+			return fmt.Errorf("the provided default value %q does not match provider type %q", defaultValue, headerType)
+		}
+	case
+		"integer":
+		_, err := strconv.ParseInt(defaultValue, 0, 64)
+		if err != nil {
+			return fmt.Errorf("the provided default value %q does not match provider type %q", defaultValue, headerType)
+		}
+	case
+		"boolean":
+		if !isBool(defaultValue) {
+			return fmt.Errorf("the provided default value %q does not match provider type %q", defaultValue, headerType)
+		}
+	}
+	return nil
+}
+
+func isString(s string) bool {
+	var digitCheck = regexp.MustCompile(regString)
+	return digitCheck.MatchString(s)
+}
+
+func isBool(s string) bool {
+	//Unable to use strconv.ParseBool because it returns truthy values https://golang.org/pkg/strconv/#example_ParseBool
+	// per https://swagger.io/docs/specification/data-models/data-types/#boolean
+	// type: boolean represents two values: true and false. Note that truthy and falsy values such as "true", "", 0 or null are not considered boolean values.
+	var digitCheck = regexp.MustCompile(regBool)
+	return digitCheck.MatchString(s)
+}
+
+func processHeaders(inputHdrs map[string]*swagger_options.Header) (swaggerHeadersObject, error) {
+	hdrs := map[string]swaggerHeaderObject{}
+	for k, v := range inputHdrs {
+		ret := swaggerHeaderObject{
+			Description: v.Description,
+			Format:      v.Format,
+			Pattern:     v.Pattern,
+		}
+		err := validateHeaderType(v.Type)
+		if err != nil {
+			return nil, err
+		}
+		ret.Type = v.Type
+
+		if v.Default != "" {
+			err := validateDefaultValueType(v.Type, v.Default)
```

Comment:
I think we need to pass in the format as well, so that we can error if a 64bit integer is used in a (integer, int32) header.

Reply:
There are cases where default is provided, but not format, or where format is provided but not default... Trying to think through all of the use cases to do this effectively. 

I will dig in more tomorrow..

Reply:
Thinking further... the format is supposed to be an extension of the type. So if the type is integer, both int32 and int64 are valid.